### DISCUSSION TITLE
account balance not update after transaction deletion (update delay?)

### DISCUSSION THREAD

**Discussion by zhiiwg** [firefly-iii/firefly-iii#6729] [General]
### problem: 

account balance not update after transaction deletion.

after delete, account balance in index page not update, and the account detail page not update, too.

### Steps to reproduce (in demo site)
1. sign in demo site. Checking Account now has 367,32 €
<img width="895" alt="image" src="https://user-images.githubusercontent.com/4954007/208665820-c0843f98-254a-415f-9014-cc20441c0476.png">

2. create a new withdrawl "test1" (367.32-100=267.32)
<img width="894" alt="image" src="https://user-images.githubusercontent.com/4954007/208666110-a52456df-dc79-4826-b0e4-53eb178e40b6.png">

3. create a new withdrawl "test2" (267.32-100=167.32)
<img width="803" alt="image" src="https://user-images.githubusercontent.com/4954007/208666591-a12f30a6-f0f4-43c3-85a7-ef150253714b.png">

5. go to transactions-expenses, delete "test1" (balance now should be 267.32)
<img width="913" alt="image" src="https://user-images.githubusercontent.com/4954007/208666281-8a66dfc1-bb4f-4473-80e7-c4b27eef26ac.png">

6. go back to index page, the balance now is still 167.32, not 267.32

![image](https://user-images.githubusercontent.com/4954007/208668282-310dcab1-c214-48aa-9914-fcc487542cfd.png)

8. and account page is still 167.32, not 267.32

![image](https://user-images.githubusercontent.com/4954007/208668502-26f42d3d-048f-4e12-aa92-e3cec360ba84.png)

transaction "test1" is already deleted

![image](https://user-images.githubusercontent.com/4954007/208668525-99140e01-ba13-4a92-9167-b5d4e6ffb1ca.png)

10. after few minutes, the balance of checking account correctly shows 267.32. 

the correction takes more time in my own site with more data(I guess, because account balance not update in few minutes). so I have question:

1. after transaction deleted, if firefly-iii need to do a calculation of data to update account balance?
2. does this update action runs instantly? and how long will it take?
3. if there is more data, will update action takes more time to do? (restart firefly-iii or choose flush in admin panel can refresh account balance instantly)

**Comment by JC5** [comment]
It should clear the cache automatically but clearly it doesn't. I'll pick it up!
